Could not install WWDR certificate

<!-- Thanks for helping fastlane! Before you submit your issue, please make sure you followed our checklist and check the appropriate boxes by putting an x in the [ ]: [x] -->

### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
<!-- Please include what's happening, expected behavior, and any relevant code samples -->
Hello everyone 👋🏻

We are using circle ci with fastlane 2.211.0. 
Randomly when we do the matching certificates, the CI fails with the following error `Could not install WWDR certificate`.
I say randomly because it does not happen each time and the logs are not relevant both ways. 

Thanks in advance for your insights

##### Command executed
<!-- The command you executed on the command line that resulted in an error -->
`match`

##### Complete output when running fastlane, including the stack trace and command used
<!-- 
You can use `--capture_output` as the last command line argument for many commands to get that collected for you. Otherwise, please do it manually.
Caution: The output of `--capture_output` could contain sensitive data such as application ids, certificate ids, or email addresses. Please make sure you double check the output and replace anything sensitive you don't wish to submit in the issue 
-->
Here is the output **when the CI fails**
<details>
  <pre> 
chruby: unknown Ruby: ruby-2.6.8
Your RubyGems version (3.0.3.1) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [✔] 🚀 
+------------------------------------------+---------+--------------------------+
|                                 Used plugins                                  |
+------------------------------------------+---------+--------------------------+
| Plugin                                   | Version | Action                   |
+------------------------------------------+---------+--------------------------+
| fastlane-plugin-diawi                    | 2.1.0   | diawi                    |
| fastlane-plugin-clean_testflight_testers | 0.3.0   | clean_testflight_testers |
+------------------------------------------+---------+--------------------------+

[15:43:52]: Sending anonymous analytics information
[15:43:52]: Learn more at https://docs.fastlane.tools/#metrics
[15:43:52]: No personal or sensitive data is sent.
[15:43:52]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[15:43:53]: ----------------------------------------
[15:43:53]: --- Step: Verifying fastlane version ---
[15:43:53]: ----------------------------------------
[15:43:53]: Your fastlane version 2.211.0 matches the minimum requirement of 2.114.0  ✅
[15:43:53]: ------------------------------
[15:43:53]: --- Step: default_platform ---
[15:43:53]: ------------------------------
[15:43:53]: Driving the lane 'ios build_and_release_ci' 🚀
[15:43:53]: -----------------------------
[15:43:53]: --- Step: setup_circle_ci ---
[15:43:53]: -----------------------------
[15:43:53]: Skipping Log Path setup as FL_OUTPUT_DIR is unset
[15:43:53]: Creating temporary keychain: "fastlane_tmp_keychain".
[15:43:53]: $ security list-keychains -d user
[15:43:53]: ▸ "/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db"
[15:43:53]: Found keychain '/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[15:43:53]: Enabling match readonly mode.
[15:43:53]: ----------------------------------
[15:43:53]: --- Step: get_info_plist_value ---
[15:43:53]: ----------------------------------
[15:43:53]: ----------------------------------
[15:43:53]: --- Step: get_info_plist_value ---
[15:43:53]: ----------------------------------
[15:43:53]: ⚙️ Start building 6.0.4 Build 1
[15:43:53]: ---------------------------------------------------
[15:43:53]: --- Step: Switch to ios post_slack_message lane ---
[15:43:53]: ---------------------------------------------------
[15:43:53]: Cruising over to lane 'ios post_slack_message' 🚖
[15:43:53]: -------------------
[15:43:53]: --- Step: slack ---
[15:43:53]: -------------------
[15:43:53]: Successfully sent Slack notification
[15:43:53]: Cruising back to lane 'ios build_and_release_ci' 🚘
[15:43:53]: ---------------------------------------------------------------
[15:43:53]: --- Step: Switch to ios test_app_store_connect_api_key lane ---
[15:43:53]: ---------------------------------------------------------------
[15:43:53]: Cruising over to lane 'ios test_app_store_connect_api_key' 🚖
[15:43:53]: ---------------------------------------
[15:43:53]: --- Step: app_store_connect_api_key ---
[15:43:53]: ---------------------------------------
[15:43:53]: Cruising back to lane 'ios build_and_release_ci' 🚘
[15:43:53]: 🤝 Match for com.toto
[15:43:53]: -------------------
[15:43:53]: --- Step: match ---
[15:43:53]: -------------------
[15:43:53]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+


+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto"]                             |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[15:43:53]: Cloning remote git repo...
[15:43:53]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[15:43:54]: Checking out branch master...
[15:43:54]: 🔓  Successfully decrypted certificates repo
[15:43:54]: Installing certificate...
[15:43:54]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[15:43:55]: --------------------------------------------
[15:43:55]: --- Step: mv Appfile.tmp Appfile || true ---
[15:43:55]: --------------------------------------------
[15:43:55]: $ mv Appfile.tmp Appfile || true
[15:43:55]: ▸ mv: rename Appfile.tmp to Appfile: No such file or directory
[15:43:55]: ---------------------------------------------------------
[15:43:55]: --- Step: Switch to ios post_slack_error_message lane ---
[15:43:55]: ---------------------------------------------------------
[15:43:55]: Cruising over to lane 'ios post_slack_error_message' 🚖
[15:43:55]: -------------------
[15:43:55]: --- Step: slack ---
[15:43:55]: -------------------
[15:43:55]: Successfully sent Slack notification
[15:43:55]: Cruising back to lane 'ios build_and_release_ci' 🚘
+-----------------------------------+--------------------------------------------------------+
|                                        Lane Context                                        |
+-----------------------------------+--------------------------------------------------------+
| DEFAULT_PLATFORM                  | ios                                                    |
| PLATFORM_NAME                     | ios                                                    |
| LANE_NAME                         | ios build_and_release_ci                               |
| KEYCHAIN_PATH                     | ~/Library/Keychains/fastlane_tmp_keychain              |
| ORIGINAL_DEFAULT_KEYCHAIN         | "/Users/distiller/Library/Keychains/login.keychain-db" |
| GET_INFO_PLIST_VALUE_CUSTOM_VALUE | 1                                                      |
+-----------------------------------+--------------------------------------------------------+
[15:43:55]: Could not install WWDR certificate

+------+---------------------------------------------------+-------------+
|                            fastlane summary                            |
+------+---------------------------------------------------+-------------+
| Step | Action                                            | Time (in s) |
+------+---------------------------------------------------+-------------+
| 1    | Verifying fastlane version                        | 0           |
| 2    | default_platform                                  | 0           |
| 3    | setup_circle_ci                                   | 0           |
| 4    | get_info_plist_value                              | 0           |
| 5    | get_info_plist_value                              | 0           |
| 6    | Switch to ios post_slack_message lane             | 0           |
| 7    | slack                                             | 0           |
| 8    | Switch to ios test_app_store_connect_api_key lane | 0           |
| 9    | app_store_connect_api_key                         | 0           |
| 💥   | match                                             | 1           |
| 11   | mv Appfile.tmp Appfile || true                    | 0           |
| 12   | Switch to ios post_slack_error_message lane       | 0           |
| 13   | slack                                             | 0           |
+------+---------------------------------------------------+-------------+

[15:43:55]: fastlane finished with errors

[!] Could not install WWDR certificate

Exited with code exit status 1
 </pre>
</details>

Here is the output when **not having the issue**

<details>
<pre>
chruby: unknown Ruby: ruby-2.6.8
Your RubyGems version (3.0.3.1) has a bug that prevents `required_ruby_version` from working for Bundler. Any scripts that use `gem install bundler` will break as soon as Bundler drops support for your Ruby version. Please upgrade RubyGems to avoid future breakage and silence this warning by running `gem update --system 3.2.3`
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [✔] 🚀 
+------------------------------------------+---------+--------------------------+
|                                 Used plugins                                  |
+------------------------------------------+---------+--------------------------+
| Plugin                                   | Version | Action                   |
+------------------------------------------+---------+--------------------------+
| fastlane-plugin-diawi                    | 2.1.0   | diawi                    |
| fastlane-plugin-clean_testflight_testers | 0.3.0   | clean_testflight_testers |
+------------------------------------------+---------+--------------------------+

[10:04:59]: Sending anonymous analytics information
[10:04:59]: Learn more at https://docs.fastlane.tools/#metrics
[10:04:59]: No personal or sensitive data is sent.
[10:04:59]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[10:05:00]: ----------------------------------------
[10:05:00]: --- Step: Verifying fastlane version ---
[10:05:00]: ----------------------------------------
[10:05:00]: Your fastlane version 2.211.0 matches the minimum requirement of 2.114.0  ✅
[10:05:00]: ------------------------------
[10:05:00]: --- Step: default_platform ---
[10:05:00]: ------------------------------
[10:05:00]: Driving the lane 'ios build_and_release_ci' 🚀
[10:05:00]: -----------------------------
[10:05:00]: --- Step: setup_circle_ci ---
[10:05:00]: -----------------------------
[10:05:00]: Skipping Log Path setup as FL_OUTPUT_DIR is unset
[10:05:00]: Creating temporary keychain: "fastlane_tmp_keychain".
[10:05:00]: $ security list-keychains -d user
[10:05:00]: ▸ "/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db"
[10:05:00]: Found keychain '/Users/distiller/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[10:05:00]: Enabling match readonly mode.
[10:05:00]: ----------------------------------
[10:05:00]: --- Step: get_info_plist_value ---
[10:05:00]: ----------------------------------
[10:05:00]: ----------------------------------
[10:05:00]: --- Step: get_info_plist_value ---
[10:05:00]: ----------------------------------
[10:05:00]: ⚙️ Start building 6.0.5 Build 2
[10:05:00]: ---------------------------------------------------
[10:05:00]: --- Step: Switch to ios post_slack_message lane ---
[10:05:00]: ---------------------------------------------------
[10:05:00]: Cruising over to lane 'ios post_slack_message' 🚖
[10:05:00]: -------------------
[10:05:00]: --- Step: slack ---
[10:05:00]: -------------------
[10:05:01]: Successfully sent Slack notification
[10:05:01]: Cruising back to lane 'ios build_and_release_ci' 🚘
[10:05:01]: ---------------------------------------------------------------
[10:05:01]: --- Step: Switch to ios test_app_store_connect_api_key lane ---
[10:05:01]: ---------------------------------------------------------------
[10:05:01]: Cruising over to lane 'ios test_app_store_connect_api_key' 🚖
[10:05:01]: ---------------------------------------
[10:05:01]: --- Step: app_store_connect_api_key ---
[10:05:01]: ---------------------------------------
[10:05:01]: Cruising back to lane 'ios build_and_release_ci' 🚘
[10:05:01]: 🤝 Match for com.toto
[10:05:01]: -------------------
[10:05:01]: --- Step: match ---
[10:05:01]: -------------------
[10:05:01]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeamTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+


+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto"]                             |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[10:05:01]: Cloning remote git repo...
[10:05:01]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[10:05:02]: Checking out branch master...
[10:05:02]: 🔓  Successfully decrypted certificates repo
[10:05:02]: Installing certificate...
[10:05:02]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[10:05:03]: There are no local code signing identities found.
You can run `security find-identity -v -p codesigning fastlane_tmp_keychain` to get this output.
This Stack Overflow thread has more information: https://stackoverflow.com/q/35390072/774.
(Check in Keychain Access for an expired WWDR certificate: https://stackoverflow.com/a/35409835/774 has more info.)
[10:05:03]: Setting key partition list... (this can take a minute if there are a lot of keys installed)
[10:05:03]: security: SecItemCopyMatching: The specified item could not be found in the keychain.
[10:05:03]: Setting key partition list... (this can take a minute if there are a lot of keys installed)

+-------------------+--------------------------------------+
|                  Installed Certificate                   |
+-------------------+--------------------------------------+
| User ID           | ABCDEFGHI7                           |
| Common Name       | Apple Distribution: TOTO (ABCDEFGHI7) |
| Organisation Unit | ABCDEFGHI7                           |
| Organisation      | TOTO                                  |
| Country           | US                                   |
| Start Datetime    | 2022-03-02 08:32:09 UTC              |
| End Datetime      | 2023-03-02 08:32:08 UTC              |
+-------------------+--------------------------------------+

[10:05:03]: Installing provisioning profile...

+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
|                                                                              Installed Provisioning Profile                                                                              |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
| Parameter           | Environment Variable                            | Value                                                                                                            |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+
| App Identifier      |                                                 | com.toto                                                                                               |
| Type                |                                                 | appstore                                                                                                         |
| Platform            |                                                 | ios                                                                                                              |
| Profile UUID        | sigh_com.toto_appstore                  | 6ad3dd91-290f-43b5-a06c-8a6e5ba9586f                                                                           |
| Profile Name        | sigh_com.toto_appstore_profile-name     | match AppStore com.toto                                                                                  |
| Profile Path        | sigh_com.toto_appstore_profile-path     | /Users/distiller/Library/MobileDevice/Provisioning Profiles/6ad3dd91-290f-43b5-a06c-8a6e5ba9586f.mobileprovision |
| Development Team ID | sigh_com.toto_appstore_team-id          | ABCDEFGHI7                                                                                                       |
| Certificate Name    | sigh_com.toto_appstore_certificate-name | Apple Distribution: TOTO (ABCDEFGHI7)                                                                             |
+---------------------+-------------------------------------------------+------------------------------------------------------------------------------------------------------------------+

[10:05:03]: All required keys, certificates and provisioning profiles are installed 🙌
[10:05:03]: Setting Provisioning Profile type to 'app-store'
[10:05:03]: 🤝 Match for com.toto.Notification
[10:05:03]: -------------------
[10:05:03]: --- Step: match ---
[10:05:03]: -------------------
[10:05:03]: Successfully loaded '/Users/distiller/project/fastlane/Matchfile' 📄

+--------------+--------------------------------------------------+
|           Detected Values from './fastlane/Matchfile'           |
+--------------+--------------------------------------------------+
| git_url      | git@github.com:MyTeam/FastlaneMatch.git |
| storage_mode | git                                              |
| type         | appstore                                         |
+--------------+--------------------------------------------------+


+----------------------------------------+--------------------------------------------------+
|                                 Summary for match 2.211.0                                 |
+----------------------------------------+--------------------------------------------------+
| type                                   | appstore                                         |
| readonly                               | true                                             |
| app_identifier                         | ["com.toto.notification"]           |
| api_key                                | ********                                         |
| generate_apple_certs                   | true                                             |
| skip_provisioning_profiles             | false                                            |
| storage_mode                           | git                                              |
| git_url                                | git@github.com:MyTeam/FastlaneMatch.git |
| git_branch                             | master                                           |
| shallow_clone                          | false                                            |
| clone_branch_directly                  | false                                            |
| skip_google_cloud_account_confirmation | false                                            |
| keychain_name                          | fastlane_tmp_keychain                            |
| force                                  | false                                            |
| force_for_new_devices                  | false                                            |
| include_mac_in_profiles                | false                                            |
| include_all_certificates               | false                                            |
| force_for_new_certificates             | false                                            |
| skip_confirmation                      | false                                            |
| safe_remove_certs                      | false                                            |
| skip_docs                              | false                                            |
| platform                               | ios                                              |
| derive_catalyst_app_identifier         | false                                            |
| fail_on_name_taken                     | false                                            |
| skip_certificate_matching              | false                                            |
| skip_set_partition_list                | false                                            |
| verbose                                | false                                            |
+----------------------------------------+--------------------------------------------------+

[10:05:03]: Cloning remote git repo...
[10:05:03]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[10:05:04]: Checking out branch master...
[10:05:04]: 🔓  Successfully decrypted certificates repo
[10:05:04]: Installing certificate...
[10:05:04]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/distiller/Library/Keychains/fastlane_tmp_keychain-db
[10:05:04]: ▸ -----BEGIN CERTIFICATE-----
.......
</pre>
</details>

### Environment
We are using fastlane `2.211.0`

Currently happening to me on Github Actions. Works fine if I run it locally.

<details>

<summary>full logs</summary>

```
[06:56:26]: Sending anonymous analytics information
[06:56:26]: Learn more at https://docs.fastlane.tools/#metrics
[06:56:26]: No personal or sensitive data is sent.
[06:56:26]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[06:56:26]: ------------------------------
[06:56:26]: --- Step: default_platform ---
[06:56:26]: ------------------------------
[06:56:26]: Driving the lane 'ios productionRelease' 🚀
[06:56:26]: ----------------------
[06:56:26]: --- Step: setup_ci ---
[06:56:26]: ----------------------
[06:56:26]: Creating temporary keychain: "fastlane_tmp_keychain".
[06:56:26]: $ security list-keychains -d user
[06:56:26]: ▸ "/Users/runner/Library/Keychains/fastlane_tmp_keychain-db"
[06:56:26]: Found keychain '/Users/runner/Library/Keychains/fastlane_tmp_keychain-db' in list-keychains, adding to search list skipped
[06:56:26]: Enabling match readonly mode.
[06:56:26]: ---------------------------------------
[06:56:26]: --- Step: app_store_connect_api_key ---
[06:56:26]: ---------------------------------------
[06:56:27]: -------------------
[06:56:27]: --- Step: match ---
[06:56:27]: -------------------
[06:56:27]: Successfully loaded '/Users/runner/work/myapp-react-native/myapp-react-native/ios/fastlane/Matchfile' 📄

+----------------+----------------------------------------------------------------------------------------+
|                               Detected Values from './fastlane/Matchfile'                               |
+----------------+----------------------------------------------------------------------------------------+
| git_url        | git@github.com:aprilmintacpineda/myapp-fastlane-match.git                              |
| storage_mode   | git                                                                                    |
| type           | appstore                                                                               |
| app_identifier | ["com.aprmp.myapp", "com.aprmp.myapp.stg", "com.aprmp.myapp.dev"]                      |
| username       | ***                                                                                    |
+----------------+----------------------------------------------------------------------------------------+


+----------------------------------------+----------------------------------------------------------------------------------------+
|                                                    Summary for match 2.211.0                                                    |
+----------------------------------------+----------------------------------------------------------------------------------------+
| api_key                                | ********                                                                               |
| readonly                               | true                                                                                   |
| type                                   | appstore                                                                               |
| generate_apple_certs                   | true                                                                                   |
| skip_provisioning_profiles             | false                                                                                  |
| app_identifier                         | ["com.aprmp.myapp", "com.aprmp.myapp.stg", "com.aprmp.myapp.dev"]                      |
| username                               | ***                                                                                    |
| team_id                                | ***                                                                                    |
| storage_mode                           | git                                                                                    |
| git_url                                | git@github.com:aprilmintacpineda/myapp-fastlane-match.git                              |
| git_branch                             | master                                                                                 |
| shallow_clone                          | false                                                                                  |
| clone_branch_directly                  | false                                                                                  |
| skip_google_cloud_account_confirmation | false                                                                                  |
| keychain_name                          | fastlane_tmp_keychain                                                                  |
| force                                  | false                                                                                  |
| force_for_new_devices                  | false                                                                                  |
| include_mac_in_profiles                | false                                                                                  |
| include_all_certificates               | false                                                                                  |
| force_for_new_certificates             | false                                                                                  |
| skip_confirmation                      | false                                                                                  |
| safe_remove_certs                      | false                                                                                  |
| skip_docs                              | false                                                                                  |
| platform                               | ios                                                                                    |
| derive_catalyst_app_identifier         | false                                                                                  |
| fail_on_name_taken                     | false                                                                                  |
| skip_certificate_matching              | false                                                                                  |
| skip_set_partition_list                | false                                                                                  |
| verbose                                | false                                                                                  |
+----------------------------------------+----------------------------------------------------------------------------------------+

[06:56:27]: Cloning remote git repo...
[06:56:27]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[06:56:29]: Checking out branch master...
[06:56:29]: 🔓  Successfully decrypted certificates repo
[06:56:29]: Installing certificate...
[06:56:29]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/runner/Library/Keychains/fastlane_tmp_keychain-db
+---------------------------+-----------------------------------------------------+
|                                  Lane Context                                   |
+---------------------------+-----------------------------------------------------+
| DEFAULT_PLATFORM          | ios                                                 |
| PLATFORM_NAME             | ios                                                 |
| LANE_NAME                 | ios productionRelease                               |
| KEYCHAIN_PATH             | ~/Library/Keychains/fastlane_tmp_keychain           |
| ORIGINAL_DEFAULT_KEYCHAIN | "/Users/runner/Library/Keychains/login.keychain-db" |
+---------------------------+-----------------------------------------------------+
[06:56:30]: Could not install WWDR certificate


[!] Could not install WWDR certificate
+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | default_platform          | 0           |
| 2    | setup_ci                  | 0           |
| 3    | app_store_connect_api_key | 0           |
| 💥   | match                     | 3           |
+------+---------------------------+-------------+

[06:56:30]: fastlane finished with errors
Error: Process completed with exit code 1.
```

</details>

--- a/fastlane/lib/fastlane/features.rb
+++ b/fastlane/lib/fastlane/features.rb
@@ -5,3 +5,6 @@
 
 FastlaneCore::Feature.register(env_var: 'FASTLANE_ENABLE_BETA_DELIVER_SYNC_SCREENSHOTS',
                                description: 'Use a newly implemented screenshots synchronization logic')
+
+FastlaneCore::Feature.register(env_var: 'FASTLANE_WWDR_USE_HTTP1_AND_RETRIES',
+                               description: 'Adds --http1.1 and --retry 3 --retry-all-errors to the curl command to download WWDR certificates')

--- a/fastlane_core/lib/fastlane_core/cert_checker.rb
+++ b/fastlane_core/lib/fastlane_core/cert_checker.rb
@@ -1,6 +1,7 @@
 require 'tempfile'
 require 'openssl'
 
+require_relative 'features'
 require_relative 'helper'
 
 # WWDR Intermediate Certificates in https://www.apple.com/certificateauthority/
@@ -155,11 +156,17 @@ module FastlaneCore
       keychain = wwdr_keychain
       keychain = "-k #{keychain.shellescape}" unless keychain.empty?
 
-      require 'open3'
+      # Attempts to fix an issue installing WWDR cert tends to fail on CIs
+      # https://github.com/fastlane/fastlane/issues/20960
+      curl_extras = ""
+      if FastlaneCore::Feature.enabled?('FASTLANE_WWDR_USE_HTTP1_AND_RETRIES')
+        curl_extras = "--http1.1 --retry 3 --retry-all-errors "
+      end
 
-      import_command = "curl -f -o #{filename} #{url} && security import #{filename} #{keychain}"
+      import_command = "curl #{curl_extras}-f -o #{filename} #{url} && security import #{filename} #{keychain}"
       UI.verbose("Installing WWDR Cert: #{import_command}")
 
+      require 'open3'
       stdout, stderr, status = Open3.capture3(import_command)
       if FastlaneCore::Globals.verbose?
         UI.command_output(stdout)

--- a/fastlane_core/spec/cert_checker_spec.rb
+++ b/fastlane_core/spec/cert_checker_spec.rb
@@ -106,19 +106,38 @@ describe FastlaneCore do
         FastlaneCore::CertChecker.installed_wwdr_certificates
       end
 
-      it 'uses the correct command to import it' do
-        # We have to execute *something* using ` since otherwise we set expectations to `nil`, which is not healthy
-        `ls`
+      describe 'uses the correct command to import it' do
+        it 'with default' do
+          # We have to execute *something* using ` since otherwise we set expectations to `nil`, which is not healthy
+          `ls`
 
-        keychain = "keychain with spaces.keychain"
-        cmd = %r{curl -f -o (([A-Z]\:)?\/.+) https://www\.apple\.com/certificateauthority/AppleWWDRCAG6\.cer && security import \1 -k #{Regexp.escape(keychain.shellescape)}}
-        require "open3"
+          keychain = "keychain with spaces.keychain"
+          cmd = %r{curl -f -o (([A-Z]\:)?\/.+) https://www\.apple\.com/certificateauthority/AppleWWDRCAG6\.cer && security import \1 -k #{Regexp.escape(keychain.shellescape)}}
+          require "open3"
 
-        expect(Open3).to receive(:capture3).with(cmd).and_return(["", "", success_status])
-        expect(FastlaneCore::CertChecker).to receive(:wwdr_keychain).and_return(keychain_name)
+          expect(Open3).to receive(:capture3).with(cmd).and_return(["", "", success_status])
+          expect(FastlaneCore::CertChecker).to receive(:wwdr_keychain).and_return(keychain_name)
 
-        allow(FastlaneCore::CertChecker).to receive(:installed_wwdr_certificates).and_return(['G1', 'G2', 'G3', 'G4', 'G5'])
-        expect(FastlaneCore::CertChecker.install_missing_wwdr_certificates).to be(1)
+          allow(FastlaneCore::CertChecker).to receive(:installed_wwdr_certificates).and_return(['G1', 'G2', 'G3', 'G4', 'G5'])
+          expect(FastlaneCore::CertChecker.install_missing_wwdr_certificates).to be(1)
+        end
+
+        it 'with FASTLANE_WWDR_USE_HTTP1_AND_RETRIES feature' do
+          # We have to execute *something* using ` since otherwise we set expectations to `nil`, which is not healthy
+          `ls`
+
+          stub_const('ENV', { "FASTLANE_WWDR_USE_HTTP1_AND_RETRIES" => "true" })
+
+          keychain = "keychain with spaces.keychain"
+          cmd = %r{curl --http1.1 --retry 3 --retry-all-errors -f -o (([A-Z]\:)?\/.+) https://www\.apple\.com/certificateauthority/AppleWWDRCAG6\.cer && security import \1 -k #{Regexp.escape(keychain.shellescape)}}
+          require "open3"
+
+          expect(Open3).to receive(:capture3).with(cmd).and_return(["", "", success_status])
+          expect(FastlaneCore::CertChecker).to receive(:wwdr_keychain).and_return(keychain_name)
+
+          allow(FastlaneCore::CertChecker).to receive(:installed_wwdr_certificates).and_return(['G1', 'G2', 'G3', 'G4', 'G5'])
+          expect(FastlaneCore::CertChecker.install_missing_wwdr_certificates).to be(1)
+        end
       end
     end
   end

